proxy/envoy/config.go

```diff
@@ -390,13 +396,35 @@ func buildTCPListener(tcpConfig *TCPRouteConfig, ip string, port int, protocol m
 				baseTCPProxy,
 			},
 		}
-	default:
-		return &Listener{
-			Name:    fmt.Sprintf("tcp_%s_%d", ip, port),
-			Address: fmt.Sprintf("tcp://%s:%d", ip, port),
-			Filters: []*NetworkFilter{baseTCPProxy},
+	case model.ProtocolREDIS:
+		// Redis filter requires the cluster name to be specified
+		// as part of the filter. We extract the cluster from the
+		// TCPRoute. Since TCPRoute has only one route, we take the
+		// cluster from the first route. The moment this route array
+		// has multiple routes, we need a fallback. For the moment,
+		// fallback to base TCP.
+
+		// Unlike Mongo, Redis is a standalone filter, that is not
+		// stacked on top of tcp_proxy
+		if len(tcpConfig.Routes) == 1 {
+			return &Listener{
+				Name:    fmt.Sprintf("redis_%s_%d", ip, port),
+				Address: fmt.Sprintf("tcp://%s:%d", ip, port),
+				Filters: []*NetworkFilter{{
+					Type: both,
+					Name: REDISProxyFilter,
+					Config: &REDISProxyFilterConfig{
+						ClusterName: tcpConfig.Routes[0].Cluster,
+						StatPrefix:  "redis",
+						ConnPool: &REDISConnPool{
+							OperationTimeoutMS: int64(REDISDefaultOpTimeout),
+						},
+					},
+				}},
+			}
 		}
 	}
+	return baseListener
```

Comment:
readability nit: no reason for `baseListener` to be defined above the switch block, since it isn't used anywhere but this return statement. Either move the `baseListener` definition down, or just return the structure.